dodiscover/metrics.py

```diff
@@ -0,0 +1,66 @@
+import networkx as nx
+import numpy as np
+from sklearn.metrics import confusion_matrix
+from sklearn.preprocessing import LabelBinarizer
+
+from ._protocol import Graph
+
+
+def confusion_matrix_networks(
+    true_graph: Graph,
+    pred_graph: Graph,
+):
+    """Compute the confusion matrix comparing a predicted graph from the true graph.
+
+    Converts the graphs into adjacency matrices, which are symmetric.
+
+    Parameters
+    ----------
+    true_graph : instance of causal graph
+        The true graph.
+    pred_graph : instance of causal graph
+        The predicted graph. The predicted graph and true graph must be
+        the same type.
+
+    Returns
+    -------
+    cm : np.ndarray of shape (2, 2)
+        The confusion matrix.
+
+    See Also
+    --------
+    sklearn.metrics.confusion_matrix
+    """
+    if set(true_graph.nodes) != set(pred_graph.nodes):
+        raise RuntimeError("Both nodes should match.")
+
+    # convert graphs to adjacency graph in networkx
+    true_graph = true_graph.to_undirected()
+    pred_graph = pred_graph.to_undirected()
+
+    # get the order of the nodes
+    idx = np.argsort(true_graph.nodes)
+    other_idx = np.argsort(pred_graph.nodes)
+
+    # next convert into 2D numpy array format and make sure nodes are ordered accordingly
+    true_adj_mat = nx.to_numpy_array(true_graph)[np.ix_(idx, idx)]
+    pred_adj_mat = nx.to_numpy_array(pred_graph)[np.ix_(other_idx, other_idx)]
+
+    # ensure we are looking at symmetric graphs
+    true_adj_mat += true_adj_mat.T
```

Comment:
I don't understand why the adjacency matrix for an undirected graph would ever not be symmetric.

Reply:
Yep, good point. Removed this redundancy.